src/sagas/backend.ts

```diff
@@ -166,6 +173,79 @@ function* backendSaga(): SagaIterator {
       yield call(showWarningMessage, 'Something went wrong. Please try again.')
     }
   })
+
+  yield takeEvery(actionTypes.FETCH_GRADING_OVERVIEWS, function*() {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }))
+    const gradingOverviews = yield call(getGradingOverviews, tokens)
+    if (gradingOverviews) {
+      yield put(actions.updateGradingOverviews(gradingOverviews))
+    }
+  })
+
+  yield takeEvery(actionTypes.FETCH_GRADING, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }))
+    const id = (action as actionTypes.IAction).payload
+    const grading = yield call(getGrading, id, tokens)
+    if (grading) {
+      yield put(actions.updateGrading(id, grading))
+    }
+  })
+
+  yield takeEvery(actionTypes.SUBMIT_GRADING, function*(action) {
+    const {
+      submissionId,
+      questionId,
+      grade,
+      comment,
+      adjustment
+    } = (action as actionTypes.IAction).payload
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }))
+    const resp = yield postGrading(submissionId, questionId, grade, comment, adjustment, tokens)
+    if (resp !== null && resp.ok) {
+      yield call(showSuccessMessage, 'Saved!', 1000)
+      // Now, update the grade for the question in the Grading in the store
+      const grading: Grading = yield select((state: IState) =>
+        state.session.gradings.get(submissionId)
+      )
+      const newGrading = grading.slice().map((gradingQuestion: GradingQuestion) => {
+        if (gradingQuestion.question.id === questionId) {
+          gradingQuestion.grade = {
+            adjustment,
+            comment,
+            grade
+          }
+        }
+        return gradingQuestion
+      })
+      yield put(actions.updateGrading(submissionId, newGrading))
+    } else if (resp !== null) {
+      let errorMessage: string
+      switch (resp.status) {
+        case 400:
+          errorMessage = 'Invalid or missing parameter(s) or submission and/or question not found'
```

Comment:
These error messages are for the end user. If they're not providing useful end-user-type information I think just let them be caught in the default case.